### PR TITLE
PDT24-24 | Create logger wrapper

### PR DESCRIPTION
**Pull request by noxiousghost**
### Tasks

- [Create logger wrapper for the application](https://outsidetech.atlassian.net/browse/PDT24-24)

### Changes

- [x] Setup [winston](https://www.npmjs.com/package/winston) and [Nest-winston](https://www.npmjs.com/package/nest-winston) for creating a custom logger.
- [x] Overwritten the default nestjs logger with new winston logger.

### Notes

- Now the logging behavior of the application depends upon the APP_ENV environment variable. In production, the logs are saved in /logs/all.log and /logs/error.log files and in development, the logs are simply printed in console.

### CODE REVIEW

File: docker-compose.yml
Comment (pratham-outside) [inline docker-compose.yml:1]: 5432:5432 chai kina hatako ? eutai rakhda ni hunxa hola

File: docker-compose.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
       POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
       POSTGRES_DB: ${POSTGRES_DB}
     ports:
-      - 5432:5432
+      - 5433:5432
```
Comment (pratham-outside) [inline docker-compose.yml:9]: 5432:5432 chai kina hatako ? eutai rakhda ni hunxa hola

File: src/config/logger.config.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+		format.printf(({ level, message, timestamp }) => {
+			return `${timestamp} [${level}]: ${message}`;
+		}),
+	),
+	transports: isProduction
+		? [
+				new transports.File({ filename: 'logs/errors.log', level: 'error' }),
+				new transports.File({ filename: 'logs/all.log' }),
+			]
+		: [new transports.Console({ format: format.combine(format.colorize({ all: true })) })],
```
Comment (pratham-outside) [inline src/config/logger.config.ts:15]: Wont there be logger file for development ?

File: src/config/logger.config.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { format, transports } from 'winston';
+const isProduction = process.env.APP_ENV === 'production';
```
Comment (pratham-outside) [inline src/config/logger.config.ts:2]: After merging PDT24-25  lets use `env.APP_ENV ===APP_ENVIRONVENT.PRODUCTION`

File: src/scripts/orm.config.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ const dataBaseConfigurations = {
 	password: process.env.POSTGRES_PASSWORD,
 	database: process.env.POSTGRES_DB,
 	synchronize: false, // Should be false in production to use migrations
-	logging: true,
+	logging: process.env.APP_ENV === 'production' ? false : true,
```
Comment (pratham-outside) [inline src/scripts/orm.config.ts:12]: `env.NODE_ENV === APP_ENVIRONVENT.PRODUCTION ? false: true ` , i think this should be the logic

File: docker-compose.yml (hunk 1 above)
Comment (noxiousghost) [reply docker-compose.yml:9]: I already postgres installed in my main machine and I assume most of the people already have it too so keeping 5432 is causing port conflict.

File: src/config/logger.config.ts (hunk 2 above)
Comment (noxiousghost) [reply src/config/logger.config.ts:15]: we can check all the logs in our console in development.

**Review by arjandhakal — APPROVED** [review]

File: src/scripts/orm.config.ts (hunk 4 above)
Comment (saugatmaharjan) [inline src/scripts/orm.config.ts:12]: may be just use `!==` rather than ternary

**Pull request merged**